### PR TITLE
Fix Pub/Sub stream binder to accommodate Spring changes.

### LINKED ISSUE #1103: Fix Pub/Sub Stream binder impl
Fix breakage of pub/sub stream binder after core changes.

See this as reference: https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/pull/471

### PR DESCRIPTION
**Pull request by dzou**
This addresses the changes in Spring Core which breaks our handling default extended binding properties in Pub/Sub stream binder. The changes in this PR follow the example: spring-cloud/spring-cloud-stream-binder-kafka#471

Fixes #1103.

### CODE REVIEW

File: spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/config/PubSubBinderConfiguration.java [hunk 1]
```diff
@@ -16,14 +16,18 @@
...
+import org.springframework.boot.context.properties.source.ConfigurationPropertyName;
 import org.springframework.cloud.gcp.pubsub.PubSubAdmin;
 import org.springframework.cloud.gcp.pubsub.core.PubSubTemplate;
 import org.springframework.cloud.gcp.stream.binder.pubsub.PubSubMessageChannelBinder;
 import org.springframework.cloud.gcp.stream.binder.pubsub.properties.PubSubExtendedBindingProperties;
 import org.springframework.cloud.gcp.stream.binder.pubsub.provisioning.PubSubChannelProvisioner;
 import org.springframework.cloud.stream.binder.Binder;
+import org.springframework.cloud.stream.config.BindingHandlerAdvise.MappingsProvider;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/config/PubSubBinderConfiguration.java]: Don't forget to add your `@author Daniel Zou`!

File: spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinder.java [hunk 2]
```diff
@@ -48,21 +48,20 @@
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinder.java]: Don't forget to add your `@author Daniel Zou` tag!

File: spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/properties/PubSubConsumerProperties.java [hunk 3]
```diff
@@ -16,12 +16,10 @@
 
 package org.springframework.cloud.gcp.stream.binder.pubsub.properties;
 
-import org.springframework.cloud.stream.config.MergableProperties;
-
 /**
  * @author João André Martins
  */
-public class PubSubConsumerProperties implements MergableProperties {
+public class PubSubConsumerProperties {
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/properties/PubSubConsumerProperties.java]: Don't forget to add your `@author Daniel Zou` tag!

File: spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/properties/PubSubExtendedBindingProperties.java [hunk 4]
```diff
@@ -16,71 +16,20 @@
...
 
 /**
  * @author João André Martins
  * @author Artem Bilan
  */
 @ConfigurationProperties("spring.cloud.stream.gcp.pubsub")
-public class PubSubExtendedBindingProperties
-		implements ExtendedBindingProperties<PubSubConsumerProperties, PubSubProducerProperties> {
+public class PubSubExtendedBindingProperties extends
+		AbstractExtendedBindingProperties<PubSubConsumerProperties, PubSubProducerProperties, PubSubBindingProperties> {
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/properties/PubSubExtendedBindingProperties.java]: Don't forget to add your `@author Daniel Zou` tag!

File: spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/properties/PubSubProducerProperties.java [hunk 5]
```diff
@@ -16,10 +16,8 @@
 
 package org.springframework.cloud.gcp.stream.binder.pubsub.properties;
 
-import org.springframework.cloud.stream.config.MergableProperties;
-
 /**
  * @author João André Martins
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/properties/PubSubProducerProperties.java]: Don't forget to add your `@author Daniel Zou` tag!

File: spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubTestBinder.java [hunk 6]
```diff
@@ -96,7 +97,8 @@ public PubSubTestBinder(String host) {
 		PubSubTemplate pubSubTemplate = new PubSubTemplate(publisherFactory, subscriberFactory);
```
Comment (ChengyuanZhao) [inline spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubTestBinder.java]: Don't forget to add your `@author Daniel Zou` tag!

File: spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinder.java [hunk 7]
```diff
@@ -48,21 +49,20 @@
...
 		super(headersToEmbed, provisioningProvider);
 		this.pubSubTemplate = pubSubTemplate;
+		this.pubSubExtendedBindingProperties = pubSubExtendedBindingProperties;
 	}
 
 	@Override
 	protected MessageHandler createProducerMessageHandler(ProducerDestination destination,
 			ExtendedProducerProperties<PubSubProducerProperties> producerProperties,
 			MessageChannel errorChannel) {
-
```
Comment (artembilan) [inline spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinder.java]: When method specification is wrapped to several lines, I prefer to have a blank line before method body for better readability. Although I heard that Boot team prefer do not have blank lines in the method body at all...

Comment (dzou) [reply spring-cloud-gcp-pubsub-stream-binder/src/main/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinder.java]: I see. Sounds good, we can do it your way. I'm fine with either; but keeping it consistent will probably be best.

**Pull request merged**